pytket/tests/qasm_test.py

```diff
@@ -328,17 +329,24 @@ def test_output_error_modes() -> None:
         c = Circuit()
         c.add_qubit(Qubit("q"))
         circuit_to_qasm_str(c)
+
+    # Adds CV as a custom gate
+    c = Circuit(2).CV(0, 1)
+    assert c == circuit_from_qasm_str(circuit_to_qasm_str(c))
+
+    c = Circuit(2)
+    c.add_gate(OpType.ZZMax, [0, 1])
+    # adds a custom gate
+    qasm_str_qelib1 = circuit_to_qasm_str(c, header="qelib1")
+    # adds "ZZ"
+    qasm_str_hqslib1 = circuit_to_qasm_str(c, header="hqslib1")
+    assert len(qasm_str_qelib1) == 354
+    assert len(qasm_str_hqslib1) == 63
```

Comment:
This seems pretty fragile -- minor changes could break this test without the code being incorrect. Could we instead check that the strings contain certain content?

Reply:
I've updated it to confirm that the strings for the gate definition is declared, and the right gate definition is used in the "qasm_str_qelib1" case, and that "ZZ" is used in the "qasm_str_hqslib1" case.